### ISSUE TITLE
cmake - absolute vs relative paths in *.gcno files (Trac #1891)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1883]
Previously we've been generating coverage on `stheno` (Ubuntu 14.04 + CVMFS). This included `gcc-4.8.4`, `cmake-2.8.12.2`, and `lcov-1.10`. We constantly received the dreaded "unexpected end of file" errors while `lcov` was reading *.gcno files. All signs pointed to upgrading `gcc` and `lcov`, so we moved coverage builds to `yoaguai` (Ubuntu 16.04) which includes `gcc-5.4`, `cmake-3.5.1` and `lcov-1.12`.

The *.gnco files generated on `stheno` contain **absolute paths** to header files, while *.gcno files generated on `yaoguai` contain **relative paths**  to header files, where `CMakeFiles/` is the base. The former obviated the need for `lcov`'s `-b/--base-directory` option. The later requires that the `-b` option be updated.

This change may break combinations of older software. This ticket serves as documentation for that case.

Migrated from https://code.icecube.wisc.edu/ticket/1891
```json
{
    "status": "closed", 
    "changetime": "2019-01-12T00:26:47", 
    "description": "Previously we've been generating coverage on `stheno` (Ubuntu 14.04 + CVMFS). This included `gcc-4.8.4`, `cmake-2.8.12.2`, and `lcov-1.10`. We constantly received the dreaded \"unexpected end of file\" errors while `lcov` was reading *.gcno files. All signs pointed to upgrading `gcc` and `lcov`, so we moved coverage builds to `yoaguai` (Ubuntu 16.04) which includes `gcc-5.4`, `cmake-3.5.1` and `lcov-1.12`.\n\nThe *.gnco files generated on `stheno` contain **absolute paths** to header files, while *.gcno files generated on `yaoguai` contain **relative paths**  to header files, where `CMakeFiles/` is the base. The former obviated the need for `lcov`'s `-b/--base-directory` option. The later requires that the `-b` option be updated.\n\nThis change may break combinations of older software. This ticket serves as documentation for that case.", 
    "reporter": "nega", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547252807474793", 
    "component": "cmake", 
    "summary": "cmake - absolute vs relative paths in *.gcno files", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2016-10-12T20:28:18", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20161012T20:31:08: 
* nega changed status from "new" to "closed"
* nega commented:

> In changeset 2679:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2679"
> tweak the base for relative paths in the coverage target
> 
> due to changes in (i think) gcc, *.gcno files store relative paths to header
> files. this updates the coverage target.
> 
> closes #1883
> ```


* nega changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190112T00:26:47: nwhitehorn commented:

> In changeset 170339:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170339"
> tweak the base for relative paths in the coverage target
> 
> due to changes in (i think) gcc, *.gcno files store relative paths to header
> files. this updates the coverage target.
> 
> closes #1883
> ```
